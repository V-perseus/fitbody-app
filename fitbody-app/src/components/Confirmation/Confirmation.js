import React from 'react'
import { StyleSheet, View, TouchableOpacity, Text } from 'react-native'
import Modal from 'react-native-modal'
import LinearGradient from 'react-native-linear-gradient'

import globals from '../../config/globals'

const styles = StyleSheet.create({
  container: {
    width: globals.window.width,
    paddingHorizontal: 25,
    flexDirection: 'column',
    // height: globals.window.height * 0.14,
    paddingBottom: 20,
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  buttonContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  title: {
    color: globals.styles.colors.colorWhite,
    fontSize: 16,
    fontFamily: globals.fonts.primary.boldStyle.fontFamily,
    flexDirection: 'column',
    textAlign: 'center',
    marginTop: 10,
    marginBottom: 10,
  },
  activeButton: {
    flex: 1,
    height: 55,
    backgroundColor: globals.styles.colors.colorWhite,
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 8,
    marginRight: 5,
  },
  activeLabel: {
    fontSize: 18,
    fontFamily: globals.fonts.primary.semiboldStyle.fontFamily,
    color: globals.styles.colors.colorPink,
  },
  inactiveButton: {
    flex: 1,
    height: 55,
    borderColor: globals.styles.colors.colorWhite,
    borderWidth: 1,
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 8,
    marginLeft: 5,
  },
  inactiveLabel: {
    fontSize: 18,
    fontFamily: globals.fonts.primary.semiboldStyle.fontFamily,
    color: globals.styles.colors.colorWhite,
  },
})

const Confirmation = (props) => (
  <Modal
    testID="confirmation_modal"
    style={{ justifyContent: 'flex-end', margin: 0 }}
    onBackButtonPress={props.handleNo}
    backdropOpacity={0.5}
    useNativeDriver={true}
    animationIn="slideInUp"
    onModalHide={props.onHide}
    animationInTiming={600}
    animationOutTiming={0}
    isVisible={props.visible}>
    <LinearGradient style={[styles.container, props.style]} colors={[globals.styles.colors.colorPink, globals.styles.colors.colorLove]}>
      <Text style={styles.title}>{props.question}</Text>
      <View style={styles.buttonContainer}>
        <TouchableOpacity style={styles.activeButton} onPress={props.handleYes}>
          <Text style={styles.activeLabel}>{props.yesText}</Text>
        </TouchableOpacity>
        <TouchableOpacity style={styles.inactiveButton} onPress={props.handleNo}>
          <Text style={styles.inactiveLabel}>{props.noText}</Text>
        </TouchableOpacity>
      </View>
    </LinearGradient>
  </Modal>
)

export default Confirmation
