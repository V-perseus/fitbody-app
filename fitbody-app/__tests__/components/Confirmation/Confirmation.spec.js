import React from 'react'
import { fireEvent, render } from '@testing-library/react-native'

import Confirmation from '../../../src/components/Confirmation/Confirmation'

describe('Confirmation', () => {
  const noHandler = jest.fn()
  const yesHandler = jest.fn()

  let component
  beforeEach(() => {
    component = render(
      <Confirmation visible={true} handleYes={yesHandler} handleNo={noHandler} yesText="yes" noText="no" question="test" />,
    )
  })

  it('renders snapshot correctly', () => {
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render modal', async () => {
    const { getByText } = component
    expect(getByText('yes')).not.toBeNull()
    expect(getByText('no')).not.toBeNull()
    expect(getByText('test')).not.toBeNull()
  })

  it('should handle "no" press', async () => {
    const { queryByText } = component
    const btn = queryByText('no')
    fireEvent.press(btn)
    expect(noHandler).toHaveBeenCalled()
  })

  it('should handle "yes" press', async () => {
    const { queryByText } = component
    const btn = queryByText('yes')
    fireEvent.press(btn)
    expect(yesHandler).toHaveBeenCalled()
  })
})
