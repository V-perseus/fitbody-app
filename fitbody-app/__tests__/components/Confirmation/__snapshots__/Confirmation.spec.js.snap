// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirmation renders snapshot correctly 1`] = `
<Modal
  animationType="none"
  deviceHeight={null}
  deviceWidth={null}
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onBackdropPress={[Function]}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[MockFunction]}
  panResponderThreshold={4}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  statusBarTranslucent={false}
  supportedOrientations={
    [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  testID="confirmation_modal"
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    forwardedRef={[Function]}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    collapsable={false}
    deviceHeight={null}
    deviceWidth={null}
    forwardedRef={[Function]}
    hideModalContentWhileAnimating={false}
    onBackdropPress={[Function]}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    panResponderThreshold={4}
    pointerEvents="box-none"
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    statusBarTranslucent={false}
    style={
      {
        "flex": 1,
        "justifyContent": "flex-end",
        "margin": 0,
        "transform": [
          {
            "translateY": 1334,
          },
        ],
      }
    }
    supportedOrientations={
      [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
  >
    <BVLinearGradient
      colors={
        [
          4291259860,
          4294929551,
        ]
      }
      endPoint={
        {
          "x": 0.5,
          "y": 1,
        }
      }
      locations={null}
      startPoint={
        {
          "x": 0.5,
          "y": 0,
        }
      }
      style={
        [
          {
            "alignItems": "center",
            "flexDirection": "column",
            "justifyContent": "space-between",
            "paddingBottom": 20,
            "paddingHorizontal": 25,
            "width": 750,
          },
          undefined,
        ]
      }
    >
      <Text
        style={
          {
            "color": "#ffffff",
            "flexDirection": "column",
            "fontFamily": "OpenSans-Bold",
            "fontSize": 16,
            "marginBottom": 10,
            "marginTop": 10,
            "textAlign": "center",
          }
        }
      >
        test
      </Text>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#ffffff",
              "borderRadius": 8,
              "flex": 1,
              "height": 55,
              "justifyContent": "center",
              "marginRight": 5,
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              {
                "color": "#fa8dd5",
                "fontFamily": "OpenSans-Semibold",
                "fontSize": 18,
              }
            }
          >
            yes
          </Text>
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "borderColor": "#ffffff",
              "borderRadius": 8,
              "borderWidth": 1,
              "flex": 1,
              "height": 55,
              "justifyContent": "center",
              "marginLeft": 5,
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              {
                "color": "#ffffff",
                "fontFamily": "OpenSans-Semibold",
                "fontSize": 18,
              }
            }
          >
            no
          </Text>
        </View>
      </View>
    </BVLinearGradient>
  </View>
</Modal>
`;
